### ISSUE TITLE
Mixpost Container doesn't work

### ISSUE THREAD

**Issue by troykelly** [inovector/mixpost#50]
### Steps to reproduce the problem

1. Run Mixpost Lite

### Expected behaviour

Mixpost Lite would start

### Actual behaviour

Mixpost Lite doesn't start

### Detailed description

Using a compose file

## docker-compose.yaml

```yaml
---
version: "3.8"
x-default-opts: &default-opts
  logging:
    driver: gelf
    options:
      gelf-address: udp://REDACTED:12201
      gelf-compression-type: none
      tag: "{{.ImageName}}/{{.Name}}/{{.ID}}"
      labels: purpose,site
services:
  mixpost:
    <<: *default-opts
    image: inovector/mixpost:latest
    environment:
      - APP_URL
      - APP_NAME
      - APP_ENV
      - APP_KEY # Generate a base64 secret here: https://generate-random.org/laravel-key-generator
      - APP_DEBUG
      - APP_PORT
      - UID
      - GID
      - DB_CONNECTION=mysql
      - DB_HOST=mysql
      - DB_PORT=3306
      - FORWARD_DB_PORT=3306
      - DB_DATABASE=${MARIADB_DATABASE}
      - DB_USERNAME=${MARIADB_USER}
      - DB_PASSWORD=${MARIADB_PASSWORD}
      - REDIS_HOST=redis
      - REDIS_PORT=6379
      - FORWARD_REDIS_PORT=6379
      - REDIS_PASSWORD
      - QUEUE_CONNECTION=redis
      - MIXPOST_DISK=mixpost
      - MIXPOST_LOG_CHANNEL=mixpost
    networks:
      - proxy
      - backend
    labels:
      - purpose=mixpost-lite
      - site=${APP_SITE}
    deploy:
      replicas: 1
      endpoint_mode: dnsrr
      placement:
        max_replicas_per_node: 1
        constraints:
          - node.role==worker
      update_config:
        parallelism: 1
        delay: 10s
      resources:
        limits:
          cpus: "8.00"
          memory: 4G
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s
      labels:
        - traefik.enable=true
        - traefik.docker.network=proxy
        - traefik.constraint-label=traefik-public
        #----------------------------------------------- routers for: odoo --------------------------------------------------
        # http
        - traefik.http.routers.${APP_SITE}-http.rule=Host(`${APP_DOMAIN}`, `www.${APP_DOMAIN}`)
        - traefik.http.routers.${APP_SITE}-http.entrypoints=http
        - traefik.http.routers.${APP_SITE}-http.middlewares=servicests
        - traefik.http.routers.${APP_SITE}-http.middlewares=https-redirect
        - traefik.http.routers.${APP_SITE}-http.service=${APP_SITE}
        # https
        - traefik.http.routers.${APP_SITE}-https.rule=Host(`${APP_DOMAIN}`, `www.${APP_DOMAIN}`) && PathPrefix(`/`)
        - traefik.http.routers.${APP_SITE}-https.entrypoints=http3
        - traefik.http.routers.${APP_SITE}-https.service=${APP_SITE}
        - traefik.http.routers.${APP_SITE}-https.tls.certresolver=${APP_SITE_RESOLVER}
        - traefik.http.routers.${APP_SITE}-https.middlewares=gzip,limit,strip-prefix
        #====================================================== services ===========================================================
        - traefik.http.services.${APP_SITE}.loadbalancer.server.port=80
    volumes:
      - storage:/var/www/html/storage/app
      - logs:/var/www/html/storage/logs
      - type: tmpfs
        target: "/tmp"
        tmpfs:
          size: 262144000
    ulimits:
      memlock:
        soft: -1
        hard: -1
      nofile:
        soft: 65536
        hard: 65536
    restart: unless-stopped
  mysql:
    image: "mariadb:latest"
    networks:
      - backend
    environment:
      - MARIADB_DATABASE
      - MARIADB_USER
      - MARIADB_PASSWORD
      - MARIADB_ROOT_PASSWORD
      - MARIADB_AUTO_UPGRADE=1
    labels:
      - purpose=mixpost-lite
      - site=${APP_SITE}
    deploy:
      replicas: 1
      endpoint_mode: dnsrr
      placement:
        max_replicas_per_node: 1
        constraints:
          - node.role==worker
      update_config:
        parallelism: 1
        delay: 10s
      resources:
        limits:
          cpus: "8.00"
          memory: 4G
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s
    volumes:
      - "mariadb:/var/lib/mysql"
      - type: tmpfs
        target: "/tmp"
        tmpfs:
          size: 262144000
    # healthcheck:
    #   test: ["CMD", "mysqladmin", "ping", "-p ${MARIADB_ROOT_PASSWORD}"]
    #   retries: 3
    #   timeout: 5s
    restart: unless-stopped
  redis:
    image: "redis:latest"
    command: redis-server --appendonly yes --replica-read-only no
    volumes:
      - "redis:/data"
      - type: tmpfs
        target: "/tmp"
        tmpfs:
          size: 262144000
    networks:
      - backend
    labels:
      - purpose=mixpost-lite
      - site=${APP_SITE}
    deploy:
      replicas: 1
      endpoint_mode: dnsrr
      placement:
        max_replicas_per_node: 1
        constraints:
          - node.role==worker
      update_config:
        parallelism: 1
        delay: 10s
      resources:
        limits:
          cpus: "8.00"
          memory: 4G
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      retries: 3
      timeout: 5s
    restart: unless-stopped
volumes:
  storage:
    driver: glusterfs
    name: ${GLUSTER_VOLUME}
  mariadb:
    driver: glusterfs
    name: ${GLUSTER_VOLUME_MARIADB}
  redis:
    driver: glusterfs
    name: ${GLUSTER_VOLUME_REDIS}
  logs:
    driver: glusterfs
    name: ${GLUSTER_VOLUME_LOGS}
networks:
  proxy:
    name: proxy
    driver: overlay
    external: true
  backend:
    driver: overlay
```

## Environment

```text
APP_DOMAIN=lite.example.com
APP_SITE=lite-example-com
APP_SITE_RESOLVER=cf
APP_DEBUG=true
MARIADB_DATABASE=mixpostlite
MARIADB_USER=mixpostlite
MARIADB_PASSWORD=REDACTED
MARIADB_ROOT_PASSWORD=REDACTED
GLUSTER_VOLUME=com_example_storage_lite
GLUSTER_VOLUME_MARIADB=com_example_mariadb_lite
GLUSTER_VOLUME_REDIS=com_example_redis_lite
GLUSTER_VOLUME_LOGS=com_example_logs_lite

# You app name. It is usually equal to the name of your project.
APP_NAME='Mixpost Lite'

# Your application environment. If you are ready for production, set 'production'
APP_ENV=production

# Is used to encrypt and decrypt sensitive data. Generate it with: 'php artisan key:generate'
APP_KEY=base64:REDACTED

# Debug mode. If the environment is in production, we recommend setting it to 'false'
APP_DEBUG=true

# URL of your application. Important: it is not allowed to be a slash at the end of the url.
APP_URL=https://lite.example.com

# You can define the application port here (Only if you install with docker-compose)
APP_PORT=80

# The user ID in your system. You can find it with 'id -u' (Only if you install with docker-compose)
UID=1000

# The group ID in your system. You can find it with 'id -g' (Only if you install with docker-compose)
GID=1000
```

## Container Log

```text
2023-07-10T04:09:52.988187128Z 
2023-07-10T04:09:52.988216484Z    INFO  Configuration cached successfully.  
2023-07-10T04:09:52.988419223Z 
2023-07-10T04:09:53.322903824Z 
2023-07-10T04:09:53.322910770Z    INFO  Application cache cleared successfully.  
2023-07-10T04:09:53.322980959Z 
2023-07-10T04:09:53.683383277Z 
2023-07-10T04:09:53.683389551Z    INFO  Routes cached successfully.  
2023-07-10T04:09:53.683521720Z 
2023-07-10T04:09:53.716854876Z  * Starting periodic command scheduler cron
2023-07-10T04:09:53.721789761Z    ...done.
2023-07-10T04:09:53.722130965Z Mixpost has started!
2023-07-10T04:09:54.024905904Z 2023-07-10 04:09:54,024 INFO Set uid to user 0 succeeded
2023-07-10T04:09:54.028178384Z 2023-07-10 04:09:54,027 INFO supervisord started with pid 28
2023-07-10T04:09:55.032125146Z 2023-07-10 04:09:55,031 INFO spawned: 'mixpost_horizon_00' with pid 30
2023-07-10T04:09:55.035094404Z 2023-07-10 04:09:55,034 INFO spawned: 'nginx' with pid 31
2023-07-10T04:09:55.038079602Z 2023-07-10 04:09:55,037 INFO spawned: 'php-fpm' with pid 32
2023-07-10T04:09:55.412453716Z 2023-07-10 04:09:55,412 INFO exited: mixpost_horizon_00 (exit status 1; not expected)
2023-07-10T04:09:56.417050263Z 2023-07-10 04:09:56,416 INFO spawned: 'mixpost_horizon_00' with pid 47
2023-07-10T04:09:56.417372580Z 2023-07-10 04:09:56,417 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-07-10T04:09:56.417479855Z 2023-07-10 04:09:56,417 INFO success: php-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-07-10T04:09:56.807615237Z 2023-07-10 04:09:56,807 INFO exited: mixpost_horizon_00 (exit status 1; not expected)
2023-07-10T04:09:58.812825285Z 2023-07-10 04:09:58,812 INFO spawned: 'mixpost_horizon_00' with pid 52
2023-07-10T04:09:59.186014789Z 2023-07-10 04:09:59,185 INFO exited: mixpost_horizon_00 (exit status 1; not expected)
2023-07-10T04:10:02.192334893Z 2023-07-10 04:10:02,191 INFO spawned: 'mixpost_horizon_00' with pid 66
2023-07-10T04:10:02.582721920Z 2023-07-10 04:10:02,582 INFO exited: mixpost_horizon_00 (exit status 1; not expected)
2023-07-10T04:10:03.584420152Z 2023-07-10 04:10:03,583 INFO gave up: mixpost_horizon_00 entered FATAL state, too many start retries too quickly
```

## Attempt to create a user to log in with

```bash
➜  ~ docker exec -it --user 1000 eb2866b61be4 /bin/bash
I have no name!@eb2866b61be4:/var/www/html$ php artisan mixpost-auth:create

 What is the name of the new user?:
 > Operations

 What is the email address of the new user?:
 > operations@example.com

 What is the password for the new user?:
 > 


In Connection.php line 795:
                                                                                                                                                                                                                
  SQLSTATE[42S02]: Base table or view not found: 1146 Table 'mixpostlite.users' doesn't exist (Connection: mysql, SQL: select exists(select * from `users` where `email` = operations@aperim.com) as `exists`)  
                                                                                                                                                                                                                

In Connection.php line 416:
                                                                                               
  SQLSTATE[42S02]: Base table or view not found: 1146 Table 'mixpostlite.users' doesn't exist  
                                                                                               

I have no name!@eb2866b61be4:/var/www/html$ 
```

### Specifications

Image: `inovector/mixpost@sha256:50da7b45b83986aab9a2069eaa3a7939b107868b76ec9c33483b2cdcf3e8149e`

**Comment by troykelly** [comment]
Fixed with

```bash
rm -Rf /var/www/html/storage/logs/installed-do-not-delete
```

**Comment by lao9s** [comment]
@troykelly Yes, this file is created after the installation is finished. The purpose of this file is to avoid repeatable running of migrating files.

To do a fresh installation, you can remove volumes and recreate the containers.

`docker-compose down -v`
then
`docker-compose up -d`

**Comment by AaronFeldman** [comment]
Had this error following the Portainer instructions.  Fixed it by increasing the timeouts in the file. rm -Rf /var/www/html/storage/logs/installed-do-not-delete did nothing to help. Followed the portainer instructions for the Lite version.  Would dearly love to pay for the full version, but one problem after another with getting this to work in docker/portainer makes it a no-go.

**Comment by lao9s** [comment]
Hi @AaronFeldman 

Installing Mixpost requires some tech knowledge. If you encountered difficulties installing it we can do it for you for a fee. Write to me if you are interested.

**Comment by AaronFeldman** [comment]
Thanks, I did get it working.  I appreciate your offer and might take advantage of it in the future.  I do have some tech knowledge, hopefully I'll be able to get up to speed without further difficulty.

**Comment by AaronFeldman** [comment]
Dima, I might need some help. Although I did get it running I am still
somewhat struggling with the https authentication issue.  I am installing
it into an Hetzner instance and I'll install a nginx web server as well as
the program into docker containers. If I still have problems getting things
going, what would it cost to get some help?  If I did get some help from
you I would also buy the pro version of the program.

Best regards,

Aaron Feldman
